packages/gen-ai-hub/src/client/openai/openai-response.ts

```diff
@@ -0,0 +1,47 @@
+import { HttpResponse } from "@sap-cloud-sdk/http-client";
+import { OpenAiChatCompletionOutput } from "./openai-types.js";
+
+/**
+ * Open AI chat completion response.
+ */
+export class OpenAiChatCompletionResponse {
+  public data: OpenAiChatCompletionOutput
+  constructor(private rawResponse: HttpResponse,) {
+    this.data = rawResponse.data;
+  }
+
+  /**
+   * Usage of tokens in the response.
+   */
+  get usage() {
+    return this.data.usage;
+  }
+  /**
+   * Reason for stopping the model.
+   */
+  get finish_reason() {
+    return this.data.choices[0].finish_reason;
+  }
+
+  /**
+* Parses the Open AI response and returns the content.
+* @param response - The  Open AI response.
+* @param choiceIndex - The index of the choice to parse.
+* @returns The message content.
+*/
+  getContent(choiceIndex: number = 0): string | null {
+    const choices = this.data.choices;
+    if (choiceIndex < 0 || choiceIndex >= choices.length) {
+      throw new Error('Invalid choice index.');
```

Comment:
Just a few thoughts:
1. If we throw an error I would like to add the given index to the message and maybe use the phrasing "choice index out of bounds" or so.
2. Do we really want to throw an error in this case? I guess when we do it on finish reason "content_filter", we can probably do that in this case too. I think I generally prefer the more graceful exit without throwing errors (also for the finish reason), but I currently don't have a better reason than my gut feeling.